src/main/java/com/netgrif/application/engine/workflow/service/DataService.java

```diff
@@ -795,8 +817,12 @@ private Object parseFieldsValues(JsonNode jsonNode, DataField dataField) {
                     value = null;
                     break;
                 }
-                String val = node.get("value").asText();
-                value = new I18nString(val);
+                if (node.get("value").isTextual()) {
```

Comment:
the `isTextual` check may be done in the `parseI18nStringValues` method. This prevents repetition

Reply:
function isnt mine and its also used for other purposes so I think its wont do